analyses/copy_number_consensus_call/src/scripts/compare_variant_calling_updated.py

```diff
@@ -0,0 +1,290 @@
+    sys.stderr.write('$$$ Write to file ' + str(output_path) + ' was sucessful\n')
+
+
+
+################### SCRIPT BODY #################################
+
+
+
+## Define parser for the input file
+parser = argparse.ArgumentParser(description="""This script takes in 3 bed files, each from one of the 3 callers
+                                                 and find common CNVs between two files at a time.""")
+parser.add_argument('--manta', required=True,
+                    help='path to the manta file')
+parser.add_argument('--cnvkit', required=True,
+                    help='path to the cnvkit file')
+parser.add_argument('--freec', required=True,
+                    help='path to the freec file')
+parser.add_argument('--manta_cnvkit', required=True,
+                    help='path of the output consensus between manta and cnvkit')
+parser.add_argument('--manta_freec', required=True,
+                    help='path of the output consensus between manta and freec')
+parser.add_argument('--cnvkit_freec', required=True,
+                    help='path of the output consensus between cnvkit and freec')
+parser.add_argument('--sample', default='no_sample',
+                    help='sample name to use in the file')
+
+args = parser.parse_args()
+
+
+## Read in the input files as dictionaries
+manta_dict = read_input_file(args.manta)
+cnvkit_dict = read_input_file(args.cnvkit)
+freec_dict = read_input_file(args.freec)
+
+## Put the input dictionaries into a bigger dictionary
+dict_of_input_content = {'manta':manta_dict, 'cnvkit':cnvkit_dict, 'freec':freec_dict}
+
+input_file_names = list(dict_of_input_content.keys())
+
+## Put the output file paths into their own lists that is to be iterated over
+## This order is important
+output_list = [args.manta_cnvkit, args.manta_freec, args.cnvkit_freec]
+
+## Make a list of iteraby index for the MAIN for-loop below
+## This give a list as followed: [0,1,2]
+list_index = list(range(0,len(dict_of_input_content)))
+
+
+## We have 3 items in dict_of_input_content. Each item is the content of an input file
+## Make a list to store the final files
+fin_list =[]
+
+## Loop through list_index
+for j, jval in enumerate(list_index):
+    for k,kval in enumerate(list_index[j+1:]):
+
+        ## Compare 1st and 2nd file from list_of_input_contents
+        ## Then compare the 1st and 3rd
+        ## Then compare the 2nd and 3rd
+        list1 = dict_of_input_content[input_file_names[jval]]
+        list2 = dict_of_input_content[input_file_names[kval]]
+
+        ## Call the "generate_consensus" function to get consensus CNVs
+        consensus_list = generate_consensus(input_file_names[jval],list1,input_file_names[kval],list2)
+
+
+
+        ## Add the consensus list into a final list that hold the
+        ## content of 3 files - manta_cnvkit , manta_freec, and cnvkit_freec
+        fin_list += [consensus_list]
+
+
+
+## Print the output into files
+for i,ival in enumerate(fin_list):
```

Comment:
Python style is to avoid iterating with indexes when you can iterate on the actual items, and `zip` is very handy for that
```suggestion
for content, outfile in zip(fin_list, output_list):
    save_to_file(content, outfile, args.sample)
```